src/OpenTelemetry.Api/OpenTelemetry.Api.csproj

```diff
@@ -11,5 +11,6 @@
 
   <ItemGroup>
     <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="$(SystemDiagnosticSourcePkgVer)" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="$(MicrosoftExtensionsDependencyInjectionAbstractionsPkgVer)" />
```

Comment:
Since some of our other instrumentation packages reference the SDK, should we just consider having the SqlClient instrumentation reference it too?

Reply:
How do you guys feel about what this PR is doing?

* Adding `Microsoft.Extensions.DependencyInjection.Abstractions` to API allows it to see `IServiceCollection`
* Adding `ConfigureServices` & `ConfigureBuilder` to `TracerProviderBuilder` allows all the fundamental DI features to work with only API reference + allows us to sunset the `IDeferredTracerProviderBuilder` 🌤️

Reply:
> it is a bug that Instrumentations depend on SDK

😆 that was my thought too, but wanted to test the waters.

In some sense, I've thought about the dependencies of the SDK project on DI, Configuration, Options, etc to be a kind of known bug/issue. As I recall, it was unintentional, but the ship sailed.

> How do you guys feel about what this PR is doing?

I feel ok about this especially given the unintended stance we fell into on the SDK project. Though, I think it warrants a discussion beyond the scope of just this PR. I think we should attempt to define boundaries of what kind of things are OK for the API and SDK to depend on going forward. It would help this discussion and the one here https://github.com/open-telemetry/opentelemetry-dotnet/pull/3639#discussion_r965407050.

Reply:
SDK depending on additional things is one thing. Its less concerning.
API can be used by library authors to instrument their libraries. (in case of .NET, a lot of libraries just need DS dependency, but some would need OTel API as well). Adding more dependency to API require a stronger justification...